### PR TITLE
Fix data access and redirects

### PR DESCRIPTION
**Pull request by JR-1991**
**Overview**

This pull request fixes the `get_datafile` endpoint and allows redirects upon requests, which is happening in the case of data access. Before this PR, the `get_datafile` method used the persistent ID access URL if `pid=False` and vice versa. This has been fixed by simply switching the `if` statements. In addition, tests have been added to pyDataverse to check if data access is working.

**TLDR**

* Fixed `get_datafile` method of `DataAccessApi` due to wrong conditions
* Redirects are now followed if a request encounters one
* Added tests for `get_datafile` (pid and id case)

### CODE REVIEW

**Review by pdurbin — APPROVED** [review]
Tests are passing. Looks like a straightforward change. I didn't test it but I'm approving this.
